src/components/ScanOrderItemModal.vue

```diff
@@ -0,0 +1,196 @@
+  IonInput,
+  IonItem,
+  IonLabel,
+  IonThumbnail,
+  IonTitle,
+  IonToolbar,
+  modalController,
+} from "@ionic/vue";
+import { computed, defineComponent } from "vue";
+import { cameraOutline, closeOutline, copyOutline, saveOutline } from "ionicons/icons";
+import { translate } from "@hotwax/dxp-components";
+import { mapGetters } from 'vuex';
+import { getProductIdentificationValue, DxpShopifyImg, useProductIdentificationStore } from '@hotwax/dxp-components';
+import { getFeature, showToast } from "@/utils"
+import Scanner from "@/components/Scanner.vue"
+
+export default defineComponent({
+  name: "ScanOrderItemModal",
+  components: { 
+    DxpShopifyImg,
+    IonButton,
+    IonButtons,
+    IonCheckbox,
+    IonContent,
+    IonFab,
+    IonFabButton,
+    IonHeader,
+    IonIcon,
+    IonInput,
+    IonItem,
+    IonLabel,
+    IonThumbnail,
+    IonTitle,
+    IonToolbar,
+  },
+  computed: {
+    ...mapGetters({
+      getProduct: 'product/getProduct',
+    }),
+  },
+  data() {
+    return {
+      orderItems: [] as any,
+      queryString: "",
+      lastScannedId: ""
+    }
+  },
+  props: ["order"],
+  mounted() {
+    this.orderItems = this.order.orderItems.length ? JSON.parse(JSON.stringify(this.order.orderItems)) : []
+  },
+  methods: {
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    async scan() {
+      const modal = await modalController.create({
+        component: Scanner,
+      });
+
+      modal.onDidDismiss()
+      .then((result) => {
+        if (result.role) {
+          this.updateProductCount(result.role);
+        }
+      })
+
+      modal.present();
+    },
+    async updateProductCount(payload: any) {
+      if(!payload) payload = this.queryString
+
+      const item = this.orderItems.find((orderItem: any) => orderItem.productSku === payload && !orderItem.isChecked);
+
+      if(item) {
+        item.isChecked = true;
+        showToast(translate("Scanned successfully.", { itemName: payload }))
+
+        this.lastScannedId = item.orderItemSeqId
+        // Highlight specific element
+        const scannedElement = document.getElementById(item.orderItemSeqId);
+        scannedElement && (scannedElement.scrollIntoView());
+
+        // Scanned product should get un-highlighted after 3s for better experience hence adding setTimeOut
+        setTimeout(() => {
+          this.lastScannedId = ''
+        }, 3000)
+      } else {
+        const item = this.orderItems.find((orderItem: any) => orderItem.productSku === payload);
+
+        if(item) {
+          showToast(translate("Product is already received:", { itemName: payload }))
+        } else {
+          showToast(translate("Scanned item is not present within the shipment:", { itemName: payload }))
+        }
+      }
+    },
+    areAllItemSelected() {
+      return !this.orderItems.some((item: any) => !item.isChecked)
+    },
+    packOrder() {
+      modalController.dismiss({ dismissed: true, packOrder: true})
```

Comment:
We can use the closeModal method and pass `packOrder` property, as using multiple methods for performing same kind of operations leads to confusion.